### PR TITLE
incremental_load.sh.tpl: Use standard tar syntax.

### LINKED ISSUE #88: Remove requirement on GNU tar
A patch in #87 explicitly made GNU tar a requirement.

The root cause of the issue that prompted the change is tar usage within [incremental_load.sh.tpl](https://github.com/bazelbuild/rules_docker/blob/master/docker/incremental_load.sh.tpl).  `tar --create --absolute-names --dereference ...` causes errors when the BSD variant is used (default on Macs).

It would be good to remove the prerequisite on GNU tar within the script so other implementations (especially BSD) could be used without users having to do anything.

Implementing #72 will most likely supersede this.

### PR DESCRIPTION
**Pull request by ohler**
Fixes #88.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: docker/incremental_load.sh.tpl [hunk 1]
```diff
@@ -214,7 +204,7 @@ EOF
   # We minimize reads / writes by symlinking the layers above
   # and then streaming exactly the layers we've established are
   # needed into the Docker daemon.
-  tar --create --absolute-names --dereference \
+  tar cPh \
```
Comment (mattmoor) [inline docker/incremental_load.sh.tpl]: nit: this will fit on a single line now.

**Comment by mattmoor** [comment]
Test this please

(For Bazel CI)

**Comment by mattmoor** [comment]
Hmm, that was easier than I thought it would be.  @cwoodcock Can you confirm this works for you?

My bad for going with the longer flag names, I mostly did it for an abundance of clarity, but who knew it would have compat issues :(

**Comment by ohler** [comment]
My only testing of this was manual, with a few docker rules I happened to have, on Mac OS Sierra.  Is this code covered by the Travis tests?

**Comment by mattmoor** [comment]
No, I'm not aware of a good way of testing this, which is why I was hoping @cwoodcock might weigh in because he added some of this logic.

**Comment by cwoodcock** [comment]
Sorry for the delay, been out and about.

Nice fix @ohler.  I should have dug deeper into how to rewrite the flags.

Not covered at the moment afaik.  But Travis does have an [OSX environment](https://docs.travis-ci.com/user/osx-ci-environment/) so hopefully it will be possible to reconfig the tests to allow that to happen.  Will do some discovery and feedback onto an issue rather than here.

Will manually test+confirm in the morning.

**Comment by cwoodcock** [comment]
LGTM

**Comment by mattmoor** [comment]
Thanks, guys!

**Pull request merged**